engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/validator/AnnotationValidatorExtractor.scala

```diff
@@ -0,0 +1,20 @@
+package pl.touk.nussknacker.engine.definition.validator
+
+import java.lang.annotation.Annotation
+import java.lang.reflect.Parameter
+
+import pl.touk.nussknacker.engine.api.definition.ParameterValidator
+
+import scala.reflect.ClassTag
+class AnnotationValidatorExtractor[AnnotationClass <: Annotation : ClassTag](parameterValidator: ParameterValidator) extends ValidatorExtractor {
+
+  override def extract(param: Parameter): Option[ParameterValidator] = {
+    param match {
+      case param if param.getAnnotation(toClass) != null => Some(parameterValidator)
+      case _ => None
+    }
+  }
+
+  protected def toClass: Class[AnnotationClass] =
```

Comment:
This should be done in factory method (`AnnotationValidatorExtractor.apply`). Class `AnnotationValidatorExtractor` should have explicit parameter `annotationClass: Class[_]`. It will be hard to use this extractor from java now. Also IMO we don't need to have generic `AnnotationClass`. It is overkill that will cause that if you will have in many places such wildcards: `AnnotationValidatorExtractor[_]`.